v2/api/network/customizations/load_balancer_extension_test.go

```diff
@@ -35,15 +34,105 @@ func Test_FuzzySetLoadBalancers(t *testing.T) {
 		},
 	}
 
-	rule := &network.LoadBalancers_InboundNatRule_Spec_ARM{
-		Name: "myrule1",
-		Properties: &network.InboundNatRulePropertiesFormat_ARM{
-			BackendPort:  to.Ptr(22),
-			FrontendPort: to.Ptr(23),
+	// Note that many of these fields are readonly and will not be present on the PUT
+	rawLBWithNatRules := map[string]any{
+		"name":     "mylb",
+		"id":       "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myrg/providers/Microsoft.Network/loadBalancers/mylb",
+		"etag":     "W/\"e764086f-6986-403d-a7e5-7e8d99301921\"",
+		"type":     "Microsoft.Network/loadBalancers",
+		"location": "westus",
+		"properties": map[string]any{
+			"provisioningState": "Succeeded",
+			"resourceGuid":      "a54785bf-72a4-4c89-ae93-fef560df2b53",
+			"inboundNatRules": []any{
+				map[string]any{
+					"name": "myrule1",
+					"id":   "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myrg/providers/Microsoft.Network/loadBalancers/mylb/inboundNatRules/myrule1",
+					"etag": "W/\"e764086f-6986-403d-a7e5-7e8d99301921\"",
+					"type": "Microsoft.Network/loadBalancers/inboundNatRules",
+					"properties": map[string]any{
+						"backendPort":  22,
+						"frontendPort": 23,
+					},
+				},
+			},
+		},
+	}
+
+	azureInboundNatRules, err := getRawInboundNatRulesField(rawLBWithNatRules)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	err = setInboundNatRules(lb, azureInboundNatRules)
+	g.Expect(err).ToNot(HaveOccurred())
+	g.Expect(lb.Location).To(Equal(to.Ptr("westus")))
+	g.Expect(lb.Properties).ToNot(BeNil())
+	g.Expect(lb.Properties.InboundNatRules).To(HaveLen(2))
+	g.Expect(lb.Properties.InboundNatRules[0].Properties).ToNot(BeNil())
+	g.Expect(lb.Properties.InboundNatRules[0].Name).To(Equal(to.Ptr("myrule")))
+	g.Expect(lb.Properties.InboundNatRules[0].Properties.BackendPort).ToNot(BeNil())
+	g.Expect(lb.Properties.InboundNatRules[0].Properties.FrontendPort).ToNot(BeNil())
```

Comment:
Suggest making it explicit that we're doing a bunch of checks on the first rule by pulling it out into a local.

```suggestion
	rule0 := lb.Properties.InboundNatRules[0]
	g.Expect(rule0.Properties).ToNot(BeNil())
	g.Expect(rule0.Name).To(Equal(to.Ptr("myrule")))
	g.Expect(rule0.Properties.BackendPort).ToNot(BeNil())
	g.Expect(rule0.Properties.FrontendPort).ToNot(BeNil())
```

Ditto for `rule1`.

Reply:
Done